parcels/particleset.py

```diff
@@ -342,9 +343,15 @@ def execute(self, pyfunc=AdvectionRK4, endtime=None, runtime=None, dt=1.,
         next_input = self.fieldset.computeTimeChunk(time, np.sign(dt))
 
         tol = 1e-12
+        if verbose_progress is None:
+            walltime_start = time_module.time()
         if verbose_progress:
             pbar = progressbar.ProgressBar(max_value=abs(endtime - _starttime))
         while (time < endtime and dt > 0) or (time > endtime and dt < 0) or dt == 0:
+            if verbose_progress is None and time_module.time() - walltime_start > 30:
+                # Showing progressbar if runtime > 30 seconds
+                pbar = progressbar.ProgressBar(max_value=abs(endtime - _starttime))
```

Comment:
So, let's change 30 to 10, since this time diff will induce an error.

Also, could you change in both locations `progressbar.ProgressBar()` to `progressbar.ProgressBar().start()` ?
Otherwise, you miss one iteration.